src/fluree/db/method/remote/core.cljc

```diff
@@ -1,41 +1,111 @@
-                   {:resource commit-key}
-                   {:keywordize-keys keywordize-keys?}))
-
-;; NOTE, below function works in conjunction with message broadcasting (not in current PR)
-#_(defn remote-read
-    "Returns a core async channel with value of remote resource."
-    [state servers commit-key keywordize-keys?]
-    (log/debug "[remote conn] remote read initiated for: " commit-key)
-    (if-let [cached (get-in @state [:resource commit-key])]
-      (go cached)
-      (xhttp/post-json (str (pick-server servers) "/fluree/remoteResource")
-                       {:resource commit-key}
-                       {:keywordize-keys keywordize-keys?})))
-
-
-;; NOTE, below function works in conjunction with message broadcasting (not in current PR)
-#_(defn remote-lookup
-    [state servers ledger-address]
-    (go-try
-      (or (get-in @state [:lookup ledger-address])
-          (let [head-commit  (<? (remote-read state servers ledger-address false))
-                head-address (get head-commit "address")]
-            (swap! state assoc-in [:lookup ledger-address] head-address)
-            (swap! state assoc-in [:resource head-address] head-commit)
-            head-address))))
\ No newline at end of file
+  (let [server-host (pick-server server-state)]
+    (xhttp/post-json (str server-host "/fluree/remoteResource")
+                     {:resource commit-key}
+                     {:keywordize-keys keywordize-keys?})))
+
+(defn monitor-messages
+  [conn msg-in msg-out]
+  (async/go-loop [next-msg (async/<! msg-in)]
+    (if next-msg
+      (let []
+        (log/warn "MESSAGE RECEIVED!!: " next-msg)
+        (recur (async/<! msg-in)))
+      (log/info "Websocket messaging connection closed."))))
+
+
+
+(defn ws-connect
+  "Returns websocket or exception wrapped in a promise-chan."
+  [server-state msg-in msg-out]
+  (let [current-server (pick-server server-state)
+        ws-chan        (async/promise-chan)
+        url            (-> current-server
+                           (str/replace-first "http" "ws")
+                           (str "/fluree/subscribe"))
+        timeout        10000
+        close-fn       (fn []
+                         (log/warn "Websocket connection closed!"))]
+    (xhttp/try-socket url msg-in msg-out ws-chan timeout close-fn) ;; will return success or error on ws-chan
+    (async/go
+      (let [ws (async/<! ws-chan)]
+        (async/close! ws-chan)
+        (if (util/exception? ws)
+          (log/warn "Error establishing websocket connection:" ws)
+          (log/debug "Websocket connection successfully established."))))
+    ws-chan))
+
+
+(defn subscribed-ledger?
+  [{:keys [server-state] :as _conn} ledger-id]
+  (boolean
+    (get-in @server-state [:subscriptions ledger-id])))
+
+(defn record-ledger-subscription
+  [{:keys [server-state] :as _conn} ledger-id]
+  (swap! server-state assoc-in [:subscriptions ledger-id] {:subscribed-at (util/current-time-millis)}))
```

Comment:
Super duper minor, but this returns the whole server state. it probably doesn't matter, but maybe we want to return just the bit added about the specific ledger we're subscribing to.